Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1446,6 +1470,37 @@ - (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_S
     completionHandler(nil);
 }
 
+- (void)_restoreData:(NSData *)data completionHandler:(void(^)(void))completionHandler
+{
+    WTF::Persistence::Decoder decoder(span(data));
+
+    std::optional<unsigned> currentLocalStorageSerializationVersion;
+    decoder >> currentLocalStorageSerializationVersion;
+    if (!currentLocalStorageSerializationVersion) {
+        completionHandler();
+        return;
+    }
+
+    std::optional<WKWebsiteDataTypeSet> dataTypes;
+    decoder >> dataTypes;
+    if (!dataTypes || dataTypes->isEmpty()) {
+        completionHandler();
+        return;
+    }
+
```

Comment:
We should construct a `CallbackAggregator` here from the completion handler.

Reply:
Added!